### DISCUSSION TITLE
Questions regarding StackLayout AndExpand options deprecation

### DISCUSSION THREAD

**Discussion by Dreamescaper** [dotnet/maui#4273] [Q&A] [answered]
So I have a section header looking like that:
![image](https://user-images.githubusercontent.com/17177729/150644176-aff9dd3f-6bc0-44d9-950e-ef8f30b8fa47.png)
Label on the left, edit icon on the right.

Currently I implement that via smth like that:
```xaml
        <StackLayout Orientation="Horizontal">
            <Label Text="Work Experience"/>
            <Image Source="edit" HorizontalOptions="EndAndExpand"/>
        </StackLayout>
```

Do I understand correctly that now I need to use Grid, smth like that?
```xaml
        <Grid ColumnDefinitions="*, Auto">
            <Label Text="Work Experience" GridLayout.Column="0" />
            <Image Source="edit" GridLayout.Column="1"/>
        </Grid>
```

Without AndExpand options, does it make any sense to set HorizontalOptions for elements in HorizontalStackLayout (or VerticalOptions for elements in VerticalStackLayout)? Would it make any effect?

**Comment by hartez** [comment, marked as answer]
@Dreamescaper your Grid replacement is correct.  

> Without AndExpand options, does it make any sense to set HorizontalOptions for elements in HorizontalStackLayout (or VerticalOptions for elements in VerticalStackLayout)? Would it make any effect?

They have no effect, so there's not really any reason to set them.

**Reply by Dreamescaper** [reply]
Thanks for an answer!

FWIW, I would say it's not really an obvious choice to use Grid in such simple cases.

(I agree that expand options caused a lot of confusion though)
